packages/next-on-pages/src/buildApplication/processVercelFunctions/invalidFunctions.ts

```diff
@@ -7,9 +7,11 @@ import {
 	stripFuncExtension,
 } from '../../utils';
 import type { CollectedFunctions, FunctionInfo } from './configs';
-import { join, resolve } from 'path';
+import { join } from 'path';
 import type { ProcessVercelFunctionsOpts } from '.';
 
+type Opts = Pick<ProcessVercelFunctionsOpts, 'functionsDir' | 'vercelConfig'>;
```

Comment:
as for the other `Opts` type I wonder if we could find a better more meaningful name for the type 😢

Also I think you're forgot to use it below:
![Screenshot 2023-10-27 at 12 23 53](https://github.com/cloudflare/next-on-pages/assets/61631103/461fbaa0-ebc3-4c7c-9483-1a89b06d5d22)